Exiled.CustomItems/API/Extensions.cs

```diff
@@ -41,6 +41,31 @@ public static class Extensions
             SpawnLocation.InsideHid,
         };
 
+        /// <summary>
+        /// Resets the player's inventory to the provided list of items and/or customitems names, clearing any items it already possess.
+        /// </summary>
+        /// <param name="player">The player to which items will be given.</param>
+        /// <param name="newItems">The new items that have to be added to the inventory.</param>
+        public static void ResetInventory(this Player player, List<string> newItems)
+        {
+            player.ClearInventory();
```

Comment:
Why would having a custom item interfere with `Inventory.Clear()`?

Reply:
Why not, is this really matter? Also, I don't see a method for Checking item in `CustomItem.` nor a method for deleting custom item, so I think that I can remove them as they would be a normal item.

Reply:
Beacuse you've to remove those custom items (if any) from `CustomItem::InsideInventories` `HashSet`.

Reply:
I still don't know why would that matter, because If I rerember correctly the item dropped and picked up again has completly 
different `uniq` and it works, so why would it be different from just removing it normally. Other than that I can't access `CustomItem.InsideInventories` either so I can't do what you asked me to do.

Reply:
You don't really -need- to remove items from that hashset. Also no, if you drop an item and pick it up, the uniq is preserved, that's the entire premise behind how CustomItems are tracked by the API.

Reply:
Hmm, okay I guess I was wrong. Do I need to fix something or PR is ready to be merged?

Reply:
Alright, but how I do that if I can't access `CustomItem.InsideInventories`?

Reply:
I feel like at that point, ClearInventory should be changed to support CustomItems, unless I'm missing the issue?

Reply:
There are more things from `Player` class doesn't support CustomItems. For example `DropItems()` drops custom items as a normal items and this should be also fixed.

Reply:
Yeah, Exiled.API is completely separated from Exiled.CustomItems.

Reply:
> Alright, but how can I do that if I can't access `CustomItem.InsideInventories`?

`CustomItem.InsideInventories` is public, you have to iterate every item and check if it's a custom one and remove it from `item.InsideInventories` if it's so, then you can safely clear the whole inventory.

Reply:
`CustomItem.InsideInventories` still I can't access it
![image](https://user-images.githubusercontent.com/75911188/115163694-dac28e00-a0aa-11eb-80b1-fcab49ace823.png)

Reply:
It is public but it's not static

Reply:
Because, as I said above, you don't have to access to `CustomItem.InsideInventories` but to `item.InsideInventories` where `item` is the iterated custom item.

Reply:
When I make it static I need to change this method too, but I'm not sure if that's okay:
```cs
        public static bool TryGet(Inventory.SyncItemInfo item, out CustomItem customItem)
        {
            // customItem = Registered?.FirstOrDefault(tempCustomItem => tempCustomItem.InsideInventories.Contains(item.uniq));
            customItem = Registered?.FirstOrDefault(tempCustomItem => InsideInventories.Contains(item.uniq));

            return customItem != null;
        }
```

Reply:
Oh wait, I read your response, give me a sec

Reply:
```suggestion
            foreach (Inventory.SyncItemInfo item in player.Inventory.items)
            {
                if (CustomItem.TryGet(item, out CustomItem customItem)
                    customItem.InsideInventories.Remove(item);
            }

            player.ClearInventory();
```

Reply:
Okay, thanks becaue I was wondering what I should remove from the HashSet

Reply:
I've updated the code.

Reply:
Alright, sorry for not reading at the first time, it's 1:30 AM so I'm not thinking normally, lol

Reply:
Np.